quantum/debounce/sym_eager_pr.c

```diff
@@ -56,12 +56,10 @@ static void transfer_matrix_values(matrix_row_t raw[], matrix_row_t cooked[], ui
 
 // we use num_rows rather than MATRIX_ROWS to support split keyboards
 void debounce_init(uint8_t num_rows) {
-    debounce_counters = (debounce_counter_t *)malloc(num_rows * sizeof(debounce_counter_t));
-    for (uint8_t r = 0; r < num_rows; r++) {
-        debounce_counters[r] = DEBOUNCE_ELAPSED;
-    }
+    const size_t debounce_counters_size = num_rows * sizeof(debounce_counter_t);
+    debounce_counters                   = (debounce_counter_t *)malloc(debounce_counters_size);
+    memset(debounce_counters, DEBOUNCE_ELAPSED, debounce_counters_size);
```

Comment:
This `memset` can easily break if this code gets revisited later on. If the type of `debounce_counter_t` gets upgraded at the same time as the value of `DEBOUNCE_ELAPSED` becomes non-zero, both of which are abstracted in this commit and which a developer will reasonably assume can be changed without much fuss, the counters will then get initialized to the wrong value.

The previous code was good in that respect.
```suggestion
    debounce_counters = (debounce_counter_t *)malloc(num_rows * sizeof(debounce_counter_t));
    for (uint8_t r = 0; r < num_rows; r++) {
        debounce_counters[r] = DEBOUNCE_ELAPSED;
    }
```